### DISCUSSION TITLE
Failed to launch osgearth project using command line

### DISCUSSION THREAD

**Discussion by Aib-star** [omnetpp/omnetpp#1011] [Q&A] [answered]
1. concrete error：Unable to find entrance
Unable to locate the program input point ZN6google8protobuf11MessageLite15ParseFromStringERKNSt7 cxx1112basic stringlcSt11char trainslcESalcEEE on the Dynamic-link library D: omnitpp-5.6.1tools win64  mingw64  bin  ibosgEarthFeatures.dIl
2. ensured：path is ok and the project can run on ide.

**Comment by rhornig** [comment, marked as answer]
If you can run the simulation from the IDE, that means that it picks up a wrong version of google's protobuf library from somewhere on your system. Check your PATH in the command window because most likely you have a directory at the beginning of the PATH that has a different version of protobuf.

**Reply by Aib-star** [reply]
How do I know which version of protobuf the IDE is loading to ensure that the project can run normally. thank u

**Reply by Aib-star** [reply]
![image](https://github.com/omnetpp/omnetpp/assets/81603653/c446d7f1-5aa5-499a-806b-a41074605ef3)
this my path and protobuf version

**Reply by Aib-star** [reply]
![image](https://github.com/omnetpp/omnetpp/assets/81603653/8ed9bfe6-9726-4701-96f7-f5039d3315f5)
this is the project`s log in cmdenv

**Reply by rhornig** [reply]
Do the bundled OMNeT++ samples work correctly (osgEarth, osgSatellites)? This seems to be an error related to osgEarth itself (i.e. trying to load a plugin for a model, that is not available). Can your osg model be loaded by the osgViewer standalone app?

**Reply by Aib-star** [reply]
osgEarth，osgSatellites have the same problem. I have never tried using OSGViewer to independently load my project

**Reply by Aib-star** [reply]
![image](https://github.com/omnetpp/omnetpp/assets/81603653/f71a70f5-1e86-4ce8-b1b2-17e2b6781c73)
I am not sure if it would be helpful to add the path to the library loaded here. Do you know the loading library path here?

**Reply by rhornig** [reply]
I'm not sure why you are loading the osgearh library manually from the command line. That must be pulled automatically by Qtenv. But a more serious issue:

You should **NOT** run your simulation from a cmd.exe prompt. **ALWAYS** start a prompt with the mingwenv.cmd command and execute all your commands there. There is a lot of things that must be properly set up in the environment to avoid these issue (like the one you are experiencing). mingwenv is speciflcally written to isolate your simulation from the rest of the installed stuff on your machine to avoid DLL hell issues like this.

1. try the official OSG samples from a mingwenv prompt.
2. If they are working, try your simulation also from there.

**Reply by Aib-star** [reply]
Because I need to launch the Omnet++simulation from the webpage, which is to run the project's exe file. Thank you, I have already solved the problem. Your first answer is the answer. You're so amazing!

**Reply by rhornig** [reply]
Interesting... Generally what do you want to achieve? You cannot launch a windows executable from a website. That would be the biggest security hole humanity have ever invented :) What is your use-case?

**Comment by Aib-star** [comment]
I didn't say in detail. I ran exe by rehaing the back-end interface through the web page. Hahahaha. 

---- Replied Message ----
| From | Rudolf ***@***.***> |
| Date | 07/18/2023 19:40 |
| To | omnetpp/omnetpp ***@***.***> |
| Cc | Aib-star ***@***.***>,
Author ***@***.***> |
| Subject | Re: [omnetpp/omnetpp] Failed to launch osgearth project using command line (Discussion #1011) |

Interesting... Generally what do you want to achieve? You cannot launch a windows executable from a website. That would be the biggest security hole humanity have ever invented :) What is your use-case?

**Reply by rhornig** [reply]
ah ok. so you are running the exe on the server, not on the client. 

A tip: On OMNeT++ 6.0 and later you can pass a command argument to `mingwenv.cmd` that would be executed **inside** the shell. This was specifically implemented to allow launching any simulation from an outside process or a shell shorcut etc. i.e. the proper way to do it is `mingwenv.cmd myomnetsimulation.exe`

**Comment by Aib-star** [comment]
It's like this. I am currently using 5.6.1. I will try the new version in the future. 

---- Replied Message ----
| From | Rudolf ***@***.***> |
| Date | 07/18/2023 20:08 |
| To | omnetpp/omnetpp ***@***.***> |
| Cc | Aib-star ***@***.***>,
Author ***@***.***> |
| Subject | Re: [omnetpp/omnetpp] Failed to launch osgearth project using command line (Discussion #1011) |
